nixos/modules/services/networking/icecream/daemon.nix

```diff
@@ -0,0 +1,165 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.icecream.daemon;
+  user = "icecc";
+  group = "icecc";
+in {
+
+  ###### interface
+
+  options = {
+
+    services.icecream.daemon = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to run the Icecream daemon.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to automatically open the specified ports in the firewall.
+        '';
+      };
+
+      openBroadcast = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to automatically open the firewall for scheduler discovery.
+        '';
+      };
+
+      baseDir = mkOption {
+        type = types.path;
+        default = "/var/tmp/icecc-daemon";
+        description = ''
+          Maximum size in Megabytes of cache used to store compile environments of compile clients.
```

Comment:
It does not look like the right description.
Does this path really need to be customizable ? For example I expect the command will break if this contains spaces.

Reply:
Yes, I need to fix the description. Reasoning behind this was to give the user the ability to put this on a different partition, I'd be fine with removing this too. The compile environments stored there can get quite large, but size can be restricted as well. IDK, open to options.

> Does this path really need to be customizable ? For example I expect the command will break if this contains spaces.

Hm, the `type = types.path` is not enough for that? It needs extra escaping than?

Reply:
There is no automatic escaping, as escaping rules are not the same for all software.

> Reasoning behind this was to give the user the ability to put this on a different partition

On NixOS it is not much harder to define a bind mount of the partition you want to use on `/var/run/icecc-daemon`.

Anyway, you know the software better than me, so up to you.